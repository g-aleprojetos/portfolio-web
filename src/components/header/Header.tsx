import React from 'react';
import {useBackgroundContext} from 'context/background';
import {useCurrentPageContext} from 'context/routesContext';
import {ToggleSwitch} from 'components/toggleSwitch';
import {Dropdown} from 'components/dropDown';
import {useTranslation} from 'react-i18next';
import {namespaces} from 'utils/i18n/i18n.constants';
import {InternalRoutes} from 'resources/enun/InternalRoutes';
import logo from 'assets/images/logo.svg';
import 'utils/i18n';
import * as S from './Header.styles';

export const Header = () => {
  const {themeDark, handleToggle} = useBackgroundContext();
  const {currentPage, handleCurrentPage} = useCurrentPageContext();
  const {t} = useTranslation(namespaces.pages.header);
  return (
    <S.HeaderContainer data-testid={'header'}>
      <S.Navegacao onClick={() => handleCurrentPage(InternalRoutes.Home)}>
        <S.ContainerLogo>
          <S.Imagem
            data-testid="test-image-logo"
            alt="Logo"
            src={logo}
            $isActivated={currentPage === InternalRoutes.Home}
          />
        </S.ContainerLogo>
      </S.Navegacao>
      <S.Nav>
        <S.Item>
          <S.ItemContent
            data-testid={`${InternalRoutes.Home}-item-content`}
            $backgroundblack={themeDark}
            $isActivated={currentPage === InternalRoutes.Home}>
            <S.Navegacao onClick={() => handleCurrentPage(InternalRoutes.Home)}>
              <S.TextoHeader
                $backgroundblack={themeDark}
                $isActivated={currentPage === InternalRoutes.Home}>
                {t('home')}
              </S.TextoHeader>
            </S.Navegacao>
          </S.ItemContent>
        </S.Item>
        <S.Item>
          <S.ItemContent
            data-testid={`${InternalRoutes.Projects}-item-content`}
            $backgroundblack={themeDark}
            $isActivated={currentPage === InternalRoutes.Projects}>
            <S.Navegacao
              onClick={() => handleCurrentPage(InternalRoutes.Projects)}>
              <S.TextoHeader
                $backgroundblack={themeDark}
                $isActivated={currentPage === InternalRoutes.Projects}>
                {t('projects')}
              </S.TextoHeader>
            </S.Navegacao>
          </S.ItemContent>
        </S.Item>
        <S.Item>
          <S.ItemContent
            data-testid={`${InternalRoutes.About}-item-content`}
            $backgroundblack={themeDark}
            $isActivated={currentPage === InternalRoutes.About}>
            <S.Navegacao
              onClick={() => handleCurrentPage(InternalRoutes.About)}>
              <S.TextoHeader
                $backgroundblack={themeDark}
                $isActivated={currentPage === InternalRoutes.About}>
                {t('about')}
              </S.TextoHeader>
            </S.Navegacao>
          </S.ItemContent>
        </S.Item>
        <S.Item>
          <Dropdown />
        </S.Item>
        <S.Item>
          <ToggleSwitch isOn={themeDark} handleOnclick={handleToggle} />
        </S.Item>
      </S.Nav>
    </S.HeaderContainer>
  );
};
