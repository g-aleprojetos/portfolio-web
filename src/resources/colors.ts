export const colors = {
  white: '#ffffff',
  lightgray: '#d3d3d3',
  black: '#000000',
  black50per: '#00000080',
  darkgray: '#23242a',
  yellow: '#FFE879',
  blue: '#15154E',
  lightblue: '#2196F3',
};
