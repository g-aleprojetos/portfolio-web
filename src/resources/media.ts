const devices = {
  xmobile: 375,
  mobile: 425,
  tablet: 768,
  laptop: 1023,
  xlaptop: 1440,
};

export const media = {
  xmobile: `(max-width: ${devices.xmobile}px)`,
  mobile: `(max-width: ${devices.mobile}px)`,
  tablet: `(max-width: ${devices.tablet}px)`,
  laptop: `(max-width: ${devices.laptop}px)`,
  xlaptop: `(max-width: ${devices.xlaptop}px)`,
};
