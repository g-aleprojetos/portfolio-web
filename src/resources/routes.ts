export default {
  Home: '/',
  Projects: '/projects',
  About: '/about',
  Settings: '/settings',
};
