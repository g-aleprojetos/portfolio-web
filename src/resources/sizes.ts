export default {
  px1: '0.063rem',
  px2: '0.125rem',
  px3: '0.188rem',
  px5: '0.313rem',
  px8: '0.5rem',
  px13: '0.813rem',
  px21: '1.313rem',
  px34: '2.125rem',
  px55: '3.438rem',
  px89: '5.563rem',
  px144: '9rem',
  px233: '14.563rem',
  px377: '23.563rem',
  px610: '38.125rem',
  media_768: '768px',
};
