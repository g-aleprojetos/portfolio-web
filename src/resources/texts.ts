const tamanho = {
  xxsmall: 12,
  xsmall: 14,
  small: 16,
  medium: 18,
  big: 24,
};

const altura = {
  xxsmall: 18,
  xsmall: 22,
  small: 24,
  medium: 26,
  big: 34,
};

export default {tamanho, altura};
