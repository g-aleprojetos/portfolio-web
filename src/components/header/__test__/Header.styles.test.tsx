import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {colors} from 'resources/colors';
import {margin} from 'resources/margins';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../Header.styles';

describe('Header.styles', () => {
  test('ContainerLogo DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ContainerLogo />);
    expect(container).toMatchSnapshot();
  });

  test('HeaderContainer DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.HeaderContainer />);
    expect(container).toMatchSnapshot();
  });

  test('Imagem DEVE aplicar cursor como "default" QUANDO $isActivated for verdadeiro', () => {
    const {container} = render(<S.Imagem src="icon.png" $isActivated={true} />);
    expect(container.firstChild).toHaveStyleRule('cursor', 'default');
  });

  test('Imagem DEVE aplicar cursor como "pointer" QUANDO $isActivated for falso', () => {
    const {container} = render(
      <S.Imagem src="icon.png" $isActivated={false} />,
    );
    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer');
  });

  test('Item DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Item />);
    expect(container).toMatchSnapshot();
  });

  test('ItemContent DEVE aplicar a borda preta QUANDO $isActivated e $backgroundblack forem verdadeiros', () => {
    const {container} = render(
      <S.ItemContent $isActivated={true} $backgroundblack={true} />,
    );
    expect(container.firstChild).toHaveStyleRule(
      'border-bottom',
      `${margin.xxxsmall}px solid ${colors.black}`,
    );
  });

  test('ItemContent DEVE aplicar a borda transparente QUANDO $isActivated for falso', () => {
    const {container} = render(<S.ItemContent $isActivated={false} />);
    expect(container.firstChild).toHaveStyleRule(
      'border-bottom',
      `${margin.xxxsmall}px solid transparent`,
    );
  });

  test('Nav DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Nav />);
    expect(container).toMatchSnapshot();
  });

  test('Navegacao DEVE ser igual ao snapshot', () => {
    const {container} = render(
      <MemoryRouter>
        <S.Navegacao to="/" />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('TextoHeader DEVE aplicar a cor preta QUANDO $backgroundblack for verdadeiro', () => {
    const {container} = render(<S.TextoHeader $backgroundblack={true} />);
    expect(container.firstChild).toHaveStyleRule('color', colors.black);
  });

  test('TextoHeader DEVE aplicar a cor de fundo padrÃ£o QUANDO $backgroundblack for falso', () => {
    const {container} = render(<S.TextoHeader $backgroundblack={false} />);
    expect(container.firstChild).toHaveStyleRule('color', colors.background01);
  });

  test('TextoHeader DEVE aplicar o cursor como "default" QUANDO $isActivated for verdadeiro', () => {
    const {container} = render(<S.TextoHeader $isActivated={true} />);
    expect(container.firstChild).toHaveStyleRule('cursor', 'default');
  });

  test('TextoHeader DEVE aplicar o cursor como "pointer" QUANDO $isActivated for falso', () => {
    const {container} = render(<S.TextoHeader $isActivated={false} />);
    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer');
  });
});
