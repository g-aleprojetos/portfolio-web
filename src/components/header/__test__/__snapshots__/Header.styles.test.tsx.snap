// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.styles ContainerLogo DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  justify-items: center;
  align-items: center;
}

@media screen and ((max-width: 768px)) {
  .c0 {
    position: fixed;
    padding: 0px;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Header.styles HeaderContainer DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  position: fixed;
  top: 0;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 116px;
  padding-top: 16px;
  padding-left: 64px;
  box-sizing: border-box;
  z-index: 10;
}

@media screen and ((max-width: 768px)) {
  .c0 {
    justify-content: center;
    padding: 0px;
  }
}

<div>
  <header
    class="c0"
  />
</div>
`;

exports[`Header.styles Item DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Header.styles Nav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  padding-right: 32px;
  gap: 8px;
}

@media screen and ((max-width: 768px)) {
  .c0 {
    display: none;
  }
}

<div>
  <nav
    class="c0"
  />
</div>
`;

exports[`Header.styles Navegacao DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
}

<div>
  <a
    class="c0"
  />
</div>
`;
